### PR TITLE
remove flash media player

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What does this change?

Removes flash support from our media player.
## What is the value of this and can you measure success?

We have a lot of moving parts, and we're a little stick in the mud when we want to try new things.

This is in the effort of helping us loosen the stick, but should also have tiny perf gains too.
## Does this affect other platforms - Amp, Apps, etc?

Nope
## Screenshots

🔪 
## Request for comment

@akash1810 @gids @sndrs

### CODE REVIEW

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 1]
```diff
@@ -171,24 +146,18 @@ define([
...
-        initPlayButtons(document.body);
-
-        mediator.on('modules:related:loaded', initPlayButtons);
-        mediator.on('page:media:moreinloaded', initPlayButtons);
     }
 
     function enhanceVideo(el, autoplay, shouldPreroll) {
         var mediaType = el.tagName.toLowerCase(),
-            $el = bonzo(el).addClass('vjs vjs-tech-' + videojs.options.techOrder[0]),
+            $el = bonzo(el).addClass('vjs vjs-tech-html5'),
```
Comment (sndrs) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: do we need `vjs vjs-tech-html5` anymore?

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 2]
```diff
@@ -115,27 +109,12 @@ define([
             // need to explicitly set the dimensions for the ima plugin.
             player.height(bonzo(player.el()).parent().dim().height);
             player.width(bonzo(player.el()).parent().dim().width);
-
```
Comment (jamesgorrie) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: This basically said - if html5 fails, reverse the techOrder (`['flash', 'html5']`) and try again.

**Comment by sndrs** [comment]
what a lovely going away present!

⭐ 👍

**Comment by akash1810** [comment]
💥 BEST PR OF THE YEAR! 👍

File: static/src/javascripts/bootstraps/enhanced/media/main.js (hunk 1 above)
Comment (jamesgorrie) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: Nope!

**Comment by jamesgorrie** [comment]
> what a lovely going away present!

Still got a week yet - there'll be more.

**Comment by gidsg** [comment]
Just to say, this will break some old videos like: https://www.theguardian.com/travel/video/2009/mar/30/brazilian-amazon-rainforest-wildlife-holiday which only have `video/x-flv` encoding. But I'd suggest let's go ahead with this and see if we get any complaints.

**Pull request merged**